iceoryx_binding_c/source/c_wait_set.cpp

```diff
@@ -187,17 +215,104 @@ void iox_ws_detach_subscriber_event(iox_ws_t const self,
                                     iox_sub_t const subscriber,
                                     const iox_SubscriberEvent subscriberEvent)
 {
+    iox::cxx::Expects(self != nullptr);
+    iox::cxx::Expects(subscriber != nullptr);
+
     self->detachEvent(*subscriber, c2cpp::subscriberEvent(subscriberEvent));
 }
 
 void iox_ws_detach_subscriber_state(iox_ws_t const self,
                                     iox_sub_t const subscriber,
                                     const iox_SubscriberState subscriberState)
 {
+    iox::cxx::Expects(self != nullptr);
+    iox::cxx::Expects(subscriber != nullptr);
+
     self->detachState(*subscriber, c2cpp::subscriberState(subscriberState));
 }
 
 void iox_ws_detach_user_trigger_event(iox_ws_t const self, iox_user_trigger_t const userTrigger)
 {
+    iox::cxx::Expects(self != nullptr);
+    iox::cxx::Expects(userTrigger != nullptr);
+
     self->detachEvent(*userTrigger);
 }
+
+iox_WaitSetResult iox_ws_attach_client_event(const iox_ws_t self,
+                                             const iox_client_t client,
+                                             const iox_ClientEvent clientEvent,
+                                             const uint64_t eventId,
+                                             void (*callback)(iox_client_t))
+{
+    iox::cxx::Expects(self != nullptr);
+    iox::cxx::Expects(client != nullptr);
+
+    auto result = self->attachEvent(*client, c2cpp::clientEvent(clientEvent), eventId, {callback, nullptr});
+    return (result.has_error()) ? cpp2c::waitSetResult(result.get_error()) : iox_WaitSetResult::WaitSetResult_SUCCESS;
+}
+
+iox_WaitSetResult iox_ws_attach_client_event_with_context_data(iox_ws_t const self,
+                                                               iox_client_t const client,
+                                                               const ENUM iox_ClientEvent clientEvent,
+                                                               const uint64_t eventId,
+                                                               void (*callback)(iox_client_t, void*),
+                                                               void* const contextData)
+{
+    iox::cxx::Expects(self != nullptr);
+    iox::cxx::Expects(client != nullptr);
```

Comment:
shouldn't `contextData` also be checked for nullptr

Reply:
Nope, it is completely valid the `contextData` is a nullptr.